### DISCUSSION TITLE
Postgres 16 compatibility

### DISCUSSION THREAD

**Discussion by Deltaus** [rdkit/rdkit#6577] [Development]
So recently tried to build 2023_03_2 with latest Postgres 16 stable branch, but saw errors like
```
error: ‘SET_VARSIZE’ was not declared in this scope
     SET_VARSIZE(res, VARHDRSZ + len);
     ^~~~~~~~~~~
```
when building and
```
undefined symbol: Abs
```
when trying to create extension and run regression tests.

First one is because postgres community split postgres.h and moved that part to a new header file varatt.h. The second one is probably because they removed Abs function?

Can anyone work on this to make it PG16 compatible? Thanks !!

**Comment by ptosco** [comment]
@Deltaus I will look into that later today, thanks for reporting.

**Comment by ptosco** [comment]
@Deltaus With [this small patch](https://patch-diff.githubusercontent.com/raw/rdkit/rdkit/pull/6580.diff) I could build the cartridge against PostgreSQL `16beta2`. Could you please check that it works for you too? Thanks in advance.

**Reply by Deltaus** [reply]
Just tested, it worked! I can create extension and run regression tests now. Thank you!
